src/Tearing/TearingEngine.inl

```diff
@@ -840,38 +813,28 @@ inline void TearingEngine<DataTypes>::calculate_inverse_distance_weights(std::ve
 
      helper::ReadAccessor< Data<VecCoord> > x(d_input_positions);
      
-     const Coord p = x[(int)vertex];
+     const Coord p = x[vertexId];
 
-    for (unsigned int tri=0 ; tri< ValidTrianglesAround.size(); tri++)
+    for (TriangleID valTriId : ValidTrianglesAround)
     {
-        
+        const Triangle& valTri = triangles[valTriId];
         sofa::type::vector<Index> VertexIndicies(numVertices);
-        
-        VertexIndicies[0] = (triangles[ValidTrianglesAround[tri]])[0];
-        VertexIndicies[1] = (triangles[ValidTrianglesAround[tri]])[1];
-        VertexIndicies[2] = (triangles[ValidTrianglesAround[tri]])[2];
-       
 
-        Coord baryCenter = (x[(int)VertexIndicies[0]]+ x[(int)VertexIndicies[1]]+ x[(int)VertexIndicies[2]]) / 3.0;
-               
+        Coord baryCenter = (x[(int)valTri[0]]+ x[(int)valTri[1]]+ x[(int)valTri[2]]) / 3.0;
```

Comment:
I do not remember why I have put this casting the index type to integer (I may have probably seen it in other codes! ). So, here I guess there is no need to have this casting.

Reply:
you are right